### DISCUSSION TITLE
How could I get AUC from survival model in package "randomForestSRC"

### DISCUSSION THREAD

**Discussion by doantrongtuyen** [kogalur/randomForestSRC#116] [Q&A] [answered]
Dear professor!
 
I’m Tuyến, I come from Vietnam (in South East of Asia). I’m using function rsfrc in your Package to fit a survival model, could you let me know that **How could I use  function  "get.auc" to get AUC from RandomSurvivalForest model IN WHICH formula Survt(time, status) is used**?
 
Thank you very much!

**Comment by ishwaran** [comment]
The following illustrates how to get the C-index directly from a survival analysis using the built in function `get.cindex`.  The C-index is related to the AUC in survival, however it accommodates the censoring that is present in survival data.  Think of C-index as being 1-AUC where AUC is "survival AUC".

```
> data(pbc, package = "randomForestSRC")
> o <- rfsrc(Surv(days, status) ~ ., pbc)
> get.cindex(o$yvar[,1], o$yvar[,2], o$predicted.oob)
[1] 0.173116
```
Notice that the value returned is exactly the same as the output:

```
> o
                         Sample size: 276
                    Number of deaths: 111
                     Number of trees: 500
           Forest terminal node size: 15
       Average no. of terminal nodes: 14.064
No. of variables tried at each split: 5
              Total no. of variables: 17
       Resampling used to grow trees: swor
    Resample size used to grow trees: 174
                            Analysis: RSF
                              Family: surv
                      Splitting rule: logrank *random*
       Number of random split points: 10
                    (OOB) Error rate: 17.31159835%

```

**Reply by doantrongtuyen** [reply]
Thank you very much for your help. Your package's wonderful!

**Reply by doantrongtuyen** [reply]
I have one additional question. Could I get AUC from survival model in which AUC is depend on time? For example AUC in following time interval [0, 100], [0, 1000] or [0, 2795]?

**Comment by ishwaran** [comment, marked as answer]
That's not possible with the `get.cindex` function, however if you are interested in survival behavior over time, I would recommend using the Brier score.  See the vignette [random survival forests](https://luminwin.github.io/randomForestSRC/articles/survival.html) and the illustration titled [Plot the Brier Score](https://luminwin.github.io/randomForestSRC/articles/survival.html).

**Reply by doantrongtuyen** [reply]
Thank you very much !
